File: smartmoneyserver/src/main/java/fpt/aptech/server/enums/transaction/TransactionSourceType.java
package fpt.aptech.server.enums.transaction;

import lombok.Getter;

/**
 * Định nghĩa nguồn gốc tạo ra một giao dịch.
 * <p>
 * Dùng cho trường {@code tTransactions.source_type} để biết giao dịch
 * được người dùng nhập tay hay do AI tạo ra.
 */
@Getter
public enum TransactionSourceType {
    MANUAL(1),          // Người dùng nhập thủ công trên form
    CHAT(2),            // Người dùng chat với AI để thêm (VD: "chi 50k cafe")
    VOICE(3),           // Người dùng dùng giọng nói để thêm
    RECEIPT(4),         // Giao dịch được tạo tự động sau khi quét hóa đơn (OCR)
    PLANNED(5);         // Giao dịch được tạo tự động từ PlannedTransaction (Scheduler hoặc Pay Bill)

    private final int value;

    TransactionSourceType(int value) {
        this.value = value;
    }
}

File: APP MOBILE/smart_money/lib/modules/transaction/enums/transaction_source_type.dart
/// Nguồn gốc tạo giao dịch: manual, chat AI, voice, receipt, scheduled
enum TransactionSourceType {
  manual,   // 1 - Nhập thủ công
  chat,     // 2 - Chat AI
  voice,    // 3 - Giọng nói
  receipt,  // 4 - Quét hóa đơn
  planned;  // 5 - Giao dịch dự kiến

  int get value => index + 1;

  static TransactionSourceType fromValue(int value) {
    if (value < 1 || value > 5) {
      return TransactionSourceType.manual;
    }
    return TransactionSourceType.values[value - 1];
  }

  String get displayName {
    switch (this) {
      case TransactionSourceType.manual:
        return 'Manual entry';
      case TransactionSourceType.chat:
        return 'Chat AI';
      case TransactionSourceType.voice:
        return 'Voice';
      case TransactionSourceType.receipt:
        return 'Scan receipt';
      case TransactionSourceType.planned:
        return 'Scheduled transaction';
    }
  }
}

